tests/io/lobster/test_inputs.py

```diff
@@ -2457,3 +2458,64 @@ def test_attributes(self):
         assert self.madelungenergies.madelungenergies_Loewdin == approx(-28.64)
         assert self.madelungenergies.madelungenergies_Mulliken == approx(-40.02)
         assert self.madelungenergies.ewald_splitting == approx(3.14)
+
+
+class TestLobsterMatrices(PymatgenTest):
+    def setUp(self) -> None:
+        self.hamilton_matrices = LobsterMatrices(
+            filename=f"{TEST_FILES_DIR}/cohp/Na_hamiltonMatrices.lobster.gz", e_fermi=-2.79650354
+        )
+        self.transfer_matrices = LobsterMatrices(filename=f"{TEST_FILES_DIR}/cohp/C_transferMatrices.lobster.gz")
+        self.overlap_matrices = LobsterMatrices(filename=f"{TEST_FILES_DIR}/cohp/Si_overlapMatrices.lobster.gz")
+        self.coeff_matrices = LobsterMatrices(filename=f"{TEST_FILES_DIR}/cohp/Si_coefficientMatricesLSO1.lobster.gz")
+
+    def test_attributes(self):
+        # hamilton matrices
+        assert self.hamilton_matrices.average_onsite_energies == {
+            "Na1_3s": 0.5885535399999997,
+            "Na1_2p_y": -25.727196460000002,
+            "Na1_2p_z": -25.727196460000002,
+            "Na1_2p_x": -25.727196460000002,
```

Comment:
Use `pytest.approx` when testing dicts containing floats. Otherwise the tests can be brittle and fail unexpectedly on different OS/Python versions.

Reply:
Thanks will keep this in mind 😃